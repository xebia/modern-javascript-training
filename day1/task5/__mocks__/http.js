module.exports = Object.assign(jest.genMockFromModule('http'), {
  get: jest.fn()
});
