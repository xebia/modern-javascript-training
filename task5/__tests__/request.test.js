jest.mock('http');

import request from '../request';
import http from 'http';

describe('the request module', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should return a promise', () => {
    const result = request();
    // expect(result).toEqual(jasmine.any(Promise));
    expect(result.then).toEqual(jasmine.any(Function));
  });

  it('should call http.get correctly', () => {
    const result = request('unittest_url');
    expect(http.get).toHaveBeenCalled();
    expect(http.get.mock.calls[0]).toMatchSnapshot();
  });
});
