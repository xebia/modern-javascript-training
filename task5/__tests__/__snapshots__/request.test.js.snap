// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the request module should call http.get correctly 1`] = `
Array [
  Object {
    "path": "unittest_url",
  },
  [Function],
]
`;
